### PR TITLE
refactor!: Deduplicate `estimateTrackParamsFromSeed` code

### PR DESCRIPTION
**Pull request by andiwand**
After a couple of iterations we can deduplicate the code with the free param estimation

blocked by
- https://github.com/acts-project/acts/pull/3835
- https://github.com/acts-project/acts/pull/3832
- https://github.com/acts-project/acts/pull/3863

### CODE REVIEW

File: Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp [hunk 1]
```diff
@@ -120,136 +117,42 @@ FreeVector estimateTrackParamsFromSeed(spacepoint_range_t spRange,
...
-  // sp in along the x axis. As such, the x-coordinate of the circle is
-  // at: x-middle / 2.
-  // The y coordinate can be found by using the straight line passing
-  // between the mid point between the middle and top sp and perpendicular to
-  // the line connecting them
-  Vector2 circleCenter;
-  circleCenter(0) = 0.5 * local1(0);
+/// @return bound parameters
+template <std::ranges::range spacepoint_range_t>
+BoundVector estimateTrackParamsFromSeed(const GeometryContext& gctx,
```
Comment (benjaminhuth) [inline Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp:127]: `Acts::Result` would be nice

File: Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp [hunk 2]
```diff
@@ -120,136 +117,42 @@ FreeVector estimateTrackParamsFromSeed(spacepoint_range_t spRange,
...
-
-  // The estimated phi and theta
   params[eBoundPhi] = VectorHelpers::phi(direction);
   params[eBoundTheta] = VectorHelpers::theta(direction);
+  params[eBoundQOverP] = freeParams[eFreeQOverP];
 
   // Transform the bottom space point to local coordinates of the provided
   // surface
-  auto lpResult = surface.globalToLocal(gctx, spGlobalPositions[0], direction);
+  auto lpResult = surface.globalToLocal(gctx, origin, direction);
```
Comment (benjaminhuth) [inline Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp:144]: directly return result here

**Review by benjaminhuth — CHANGES_REQUESTED** [review]

**Review by benjaminhuth — DISMISSED** [review]
Looks good!

**Review by benjaminhuth — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [5c3a8b81117927452c50c0947d2b47e6f76c1721]
### Build job with this PR failed!
Please investigate the [build job](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/46351355) for the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8567819)!

**Comment by paulgessinger** [comment]
So looks like this changes specifically the HGTD workflow counts in Athena.

**Pull request merged**